tests/subsys/storage/stream/stream_flash/src/main.c

```diff
@@ -643,7 +643,7 @@ static void test_stream_flash_progress_clear(void)
 #endif
 }
 
-void test_main(void)
+static void *lib_stream_flash_test_setup(void)
 {
 	__ASSERT_NO_MSG(device_is_ready(fdev));
```

Comment:
This was technically incorrect before and we should go ahead and fix it while in this file. Asserting inside `test_main` or `setup` will blow up the main thread. If you'd like to assert this it should be done in the `before` phase. Same with the assert below on the page size.

Though technically, they should be using the assume API (https://github.com/zephyrproject-rtos/zephyr/pull/46961) which still needs 1 more approval.

Reply:
I move this into before_fn, I keep the same logic as before: the device is not ready, assert fail.

Reply:
2 comments:
1. The function name should reflect it's position in the test. Please rename to use `_before` suffix
2. Please replace `__ASSERT_NO_MSG` with `zassume_true` otherwise you'll get incorrect reporting in twister (there are 2 asserts here, please update both)